test/functional/attributes.js

```diff
@@ -103,6 +103,54 @@ describe('attribute updates', () => {
       expect(el.getAttribute('data-foo')).to.equal(null);
     });
 
+    it('should keep attribute in different position', () => {
```

Comment:
> Also, one of the later tests (dropping foo from the start of an element with 3 attributes) isn't any different than the first test..

Actually, that was a hard fought bug. The difference is dropping a attribute from the front, then the last (the bug had the first reappearing, trying to optimize for updating only the changed attributes not all of them).